### ISSUE TITLE
aws_lambda_permission doesn't support update

### ISSUE THREAD

**Issue by erutherford** [hashicorp/terraform#8559]
### Terraform Version

`v0.7.2`
### Affected Resource(s)

aws_lambda_permission
### Terraform Configuration Files

``` hcl
resource "aws_lambda_permission" "invoke_thumbnail_from_sns" {
  statement_id = "AllowInvokationFromSNS"
  action = "lambda:InvokeFunction"
  count = "${length(var.devs)}"
  function_name = "${aws_lambda_function.create_thumbnail.*.arn[count.index]}"
  principal = "sns.amazonaws.com"
  source_arn = "${aws_sns_topic.asset_created.*.arn[count.index]}"
}
```
### Expected Behavior

If this resource doesn't support updates, then it should destroy/recreate the resource.
### Actual Behavior

It tried to modify the resource and failed.  I had to taint the resources manually and re-run.
### Extra information

I'm not sure if it matters, but specifically I removed a `source_account` entry in the policy and that was when I threw the doesn't support update error.
### Steps to Reproduce
1. create a `aws_lambda_permission` resource
2. apply changes
3. change the configuration of the `aws_lambda_permission` resource 
4. apply changes

**Comment by stack72** [comment]
Hi @erutherford 

Please can you tell me if this is still an issue? I have been able to successfully modify a lambda permission and force a new resource

Paul

**Comment by ryandub** [comment]
I ran into the same issue with an older version of Terraform and can confirm that at least with version `v0.8.7` a new resource is created instead of the previous error. 👍

**Comment by catsby** [comment]
Hi! I believe this was fixed in https://github.com/hashicorp/terraform/pull/9699 and released in (at least) v0.7.8 of Terraform
